deploy/gk-deploy

```diff
@@ -256,6 +256,9 @@ check() {
   if [[ ${cond} =~ $number_re ]]; then
     wait_limit=${cond}
   fi
+  if [[ "${resource}" == *s ]]; then
```

Comment:
This is a little weird at first sight.
* Firstly, does it mean you change how the second argument is used, depending on whether the first argument ends with the letter 's'? This sounds a little fragile, or implicit.
* And at the very least it should be explained in the commit message, and maybe a function header comment.
* Should `*s` be quoted? Like `"*s"`?